### PR TITLE
[Circuit][Report] Feat/report negative circuit

### LINKED ISSUE #379: [UST-165] [Circuit] Claim Negative Reputation
**Is your feature request related to a problem? Please describe.**
Implement a new circuit that proves that the input `user_id`, `reported_epoch` is able to generate `reported_epoch_key` and outputs `current_epoch_key`.

**Describe the solution you'd like**
```circom
`reportNegRepProof`
public input reported_epoch_key; // if stored on-chain
private input user_id;
private input reported_epoch;
private input current_epoch;
private input current_nonce; // action count
public output current_epoch_key;

for nonce in 0..EPOCH_KEY_NONCE:
  // check one of the reported epoch key is matched
	if reported_epoch_key == poseidon(user_id, reported_epoch, nonce)

current_epoch_key = poseidon(user_id, current_epoch, current_nonce)
```

**Todos**
- [ ] Implement a circuit for claiming negative reputation

**Tests**
- Should success with valid inputs
- Should revert with wrong 
    - `user_id`
    - `reported_epoch`

### PR DESCRIPTION
**Pull request by FoodChain1028**
## Summary

Add report negative reputation proof circuit in `circuit`

## Linked Issue

close #379 

## Details

Added a circuit for reported user/ failed reporter to prove their epochKey is theirs.

## Tests

- [X] should generate a report negative reputation proof and output with correct epochKey
- [X] should revert with invalid hashUserId
- [X] should revert with invalid reportedEpoch

## Possible Impacts

-

## Verification Steps

```
yarn circuits test
```

## Todo

Break down the progress of the PR for everyone to see what else you intend to include in this PR.

-   [X] Draft implementation for `function`
-   [X] Unit tests for `function`

## Checklist

-   [X] All new and existing tests pass
-   [X] I have updated the documentation (if applicable)
-   [X] My changes do not introduce new security vulnerabilities

### CODE REVIEW

File: packages/circuits/test/proveReportNegRep.test.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    )
+    const isValid = await defaultProver.verifyProof(
+        circuit,
+        publicSignals,
+        proof
+    )
+    return { isValid, proof, publicSignals }
+}
+
+const createRandomUserIdentity = (): IdentityObject => {
```
Comment (ryanycw) [inline packages/circuits/test/proveReportNegRep.test.ts:62]: Note: We can consider moving these common utils to a separate file.

**Review by ryanycw — APPROVED** [review]

File: packages/circuits/test/proveReportNegRep.test.ts (hunk 1 above)
Comment (ryanycw) [reply packages/circuits/test/proveReportNegRep.test.ts:62]: @FoodChain1028 See if you want to make this change in this PR or leave this in the future.

Comment (FoodChain1028) [reply packages/circuits/test/proveReportNegRep.test.ts:62]: I will make it in the future PR :)

Comment (FoodChain1028) [reply packages/circuits/test/proveReportNegRep.test.ts:62]: I just create a issue with this: #410

**Pull request merged**